src/analyses/goto_check.cpp

```diff
@@ -1333,8 +1333,17 @@ void goto_checkt::bounds_check(
     auto type_size_opt = size_of_expr(ode.root_object().type(), ns);
 
     if(type_size_opt.has_value())
-      type_matches_size =
-        equal_exprt(size, typecast_exprt(type_size_opt.value(), size.type()));
+    {
+      type_matches_size = if_exprt{
```

Comment:
This definitely warrants an explanatory comment

Reply:
Will add one as soon as #4325 is in.

Reply:
Done.